components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/AuthorizationGrantCache.java

```diff
@@ -63,13 +69,19 @@ public static AuthorizationGrantCache getInstance() {
         }
         return instance;
     }
+
     /**
      * Add a cache entry by access token.
      *
      * @param key   Key which cache entry is indexed.
      * @param entry Actual object where cache entry is placed.
      */
     public void addToCacheByToken(AuthorizationGrantCacheKey key, AuthorizationGrantCacheEntry entry) {
+
+        if (log.isDebugEnabled() && key != null && entry != null) {
+            log.debug("Adding to cache by token - " + key.getUserAttributesId());
+            logClaims("Add to cache by token", entry.getUserAttributes());
```

Comment:
Can we log user claims?
What if there are sensitive information stored in claim values?